scene/audio/audio_stream_player.h

```diff
@@ -89,6 +91,7 @@ class AudioStreamPlayer : public Node {
 	int get_max_polyphony() const;
 
 	void play(float p_from_pos = 0.0);
+	void play_scheduled(float p_abs_time, float p_from_pos = 0.0);
```

Comment:
```suggestion
	void play_scheduled(double p_abs_time, double p_from_pos = 0.0);
```

Thoughts? I would like to suggest anytime we use position as time it'll be double.

Reply:
Feels excessively precise for audio.

Reply:
Yeah this is a good idea. (300 + 1/44100) seconds in a 32 bit float has a ~0.35 audio frame error, which will only get worse as the game keeps running.

I've updated this as well as `get_playback_position`, `play`, and `seek` to use doubles. The metronome in the test project sounds much better with doubles, but that could just be placebo :)

Reply:
I am not sure how to deal with api breakage. I think the previous api is in error because https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-private-method-physics-process 's delta is a double and time as position should be a double.

```
Compatibility to 4.1-stable is broken in the following ways:

Validate extension JSON: Error: Field 'classes/AudioStreamPlayer/methods/get_playback_position/return_value': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer/methods/play/arguments/0': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer/methods/seek/arguments/0': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer2D/methods/get_playback_position/return_value': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer2D/methods/play/arguments/0': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer2D/methods/seek/arguments/0': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer3D/methods/get_playback_position/return_value': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer3D/methods/play/arguments/0': meta changed value in new API, from "float" to "double".
Validate extension JSON: Error: Field 'classes/AudioStreamPlayer3D/methods/seek/arguments/0': meta changed value in new API, from "float" to "double".
```

Reply:
I also think it's an error since `AudioStream.get_playback_position()` returns a `double`.

That being said, would it be worth separating out the API breakage into a different PR to unblock this? `double` precision for song position doesn't seem too necessary since we'd only see audio frame rounding errors at ~7 minutes, which is very much an edge case.

Reply:
Let me ask for a second opinion

Reply:
To clarify, we'd keep `p_abs_time` a double, and revert `p_from_pos` to a float.

Reply:
I think that at least modifying the new parameters to double won't be compact breaking.

Reply:
Sounds good, I'll leave the existing API functions alone and will only change the new API + engine code.